include/mbgl/style/expression/formatted.hpp

```diff
@@ -17,17 +18,19 @@ extern const char* const kFormattedSectionTextFont;
 extern const char* const kFormattedSectionTextColor;
 
 struct FormattedSection {
-    FormattedSection(std::string text_,
-                     optional<double> fontScale_,
-                     optional<FontStack> fontStack_,
-                     optional<Color> textColor_)
-        : text(std::move(text_))
-        , fontScale(std::move(fontScale_))
-        , fontStack(std::move(fontStack_))
-        , textColor(std::move(textColor_))
-    {}
+    explicit FormattedSection(std::string text_,
+                              optional<double> fontScale_,
+                              optional<FontStack> fontStack_,
+                              optional<Color> textColor_)
+        : text(std::move(text_)),
+          fontScale(std::move(fontScale_)),
+          fontStack(std::move(fontStack_)),
+          textColor(std::move(textColor_)) {}
+
+    explicit FormattedSection(Image image_) : image(std::move(image_)) {}
```

Comment:
Wouldn't be the case for having a different struct for `FormattedImageSection`?

Reply:
I tried that, it over-complicated code a lot and increased binary size. If there would be more section types than image / text, I would refactor it.